apollo-router-core/src/plugin.rs

```diff
@@ -103,7 +104,7 @@ pub trait Plugin: Send + Sync + 'static + Sized {
 }
 
 #[async_trait]
-pub trait DynPlugin: Send + Sync + 'static {
+pub trait DynPlugin: Display + Send + Sync + 'static {
```

Comment:
I think this trait bound is a little bit annoying for people who just want to implement a plugin. And also requiring Display just to display the name is a little bit weird to me. Maybe we could add a method in the plugin trait like `fn name() -> &'static str` of equivalent. It's less restrictive and pretty straightforward. And with this function in the trait we could get rid of passing the name in the `register_plugin` macro. What do you think ?